core/services/pipeline/scheduler_test.go

```diff
@@ -0,0 +1,171 @@
+package pipeline
```

Comment:
I'm testing `scheduler` which is a private type so the code had to sit in the `pipeline` package

Reply:
```
(pipeline/helpers_test.go)
package pipeline

type Scheduler = scheduler
```
This is the trick you want

Reply:
whats wrong with having a whitebox test though? Seems fine to just have a scheduler_test.go in the pipeline package